Objects/unicodeobject.c

```diff
@@ -13021,40 +13021,41 @@ unicode_zfill_impl(PyObject *self, Py_ssize_t width)
     return u;
 }
 
-PyDoc_STRVAR(startswith__doc__,
-             "S.startswith(prefix[, start[, end]]) -> bool\n\
-\n\
-Return True if S starts with the specified prefix, False otherwise.\n\
-With optional start, test S beginning at that position.\n\
-With optional end, stop comparing S at that position.\n\
-prefix can also be a tuple of strings to try.");
+/*[clinic input]
+@text_signature "($self, prefix[, start[, end]], /)"
+str.startswith as unicode_startswith
+
+    prefix as subobj: object
+        A string or a tuple of strings to try.
+    start: slice_index(accept={int, NoneType}, c_default='0') = None
+        Optional start position. Default 0.
+    end: slice_index(accept={int, NoneType}, c_default='PY_SSIZE_T_MAX') = None
+        Optional stop position. Default PY_SSIZE_T_MAX.
```

Comment:
Does this text show up in the user-visible docstring? PY_SSIZE_T_MAX isn't very useful to Python users; the effective default is just "look to the end of the string". I would omit this part:
```suggestion
        Optional stop position.
```

Reply:
Yep, param docstrings are embedded by Argument Clinic into the "body" of the docstring.

Perhaps:

```suggestion
        Optional stop position. Default: the end of the string.
```

Reply:
BTW; you can see the generated output in `Objects/clinic/unicodeobject.c.h`.

Reply:
Your suggestion sounds good!

Ah right, should have looked at the generated output.

Reply:
Thanks for pointing this out; I think the docstring looks better now!